implementations/rust/ockam/ockam_command/src/storage.rs

```diff
@@ -0,0 +1,163 @@
+pub fn ensure_identity_exists(expect_trusted: bool) -> Result<()> {
+    let dir = get_ockam_dir()?;
+    if !dir.exists() {
+        anyhow::bail!(
+            "Failed to locate the ockam directory. Do you need to run `ockam create-identity`?"
+        );
+    }
+    if !dir.join("identity.json").exists() || !dir.join("vault.json").exists() {
+        anyhow::bail!(
+            "No identity has been initialized in the ockam directory at `{}`. \
+            You may need to need to run `ockam create-identity`. If this directory \
+            should not be used, you may use the `OCKAM_DIR` environment variable \
+            instead.",
+            dir.display(),
+        );
+    }
+    if expect_trusted && !dir.join("trusted").exists() {
+        eprintln!(
+            "warning: The ockam directory does not have a list of trusted \
+            identifiers at `{}/trusted`. This may indicate a configuration error",
+            dir.display(),
+        );
+    }
+    Ok(())
+}
+
+#[tracing::instrument(level = "debug", err)]
+pub fn init_ockam_dir() -> Result<std::path::PathBuf> {
+    use std::os::unix::fs::{DirBuilderExt, MetadataExt, PermissionsExt};
+    let path = get_ockam_dir()?;
+    tracing::debug!("Ockam dir will be at: {:?}", path);
+    let parent = path.parent();
+    anyhow::ensure!(
+        !path.is_relative() || path.parent().is_none(),
+        "Please use an absolute path if `$OCKAM_DIR` is manually specified.",
+    );
+    if !path.exists() {
+        let parent = parent.unwrap();
+        if !parent.exists() {
+            eprintln!("Creating parent of ockam directory at: {:?}", path);
+            std::fs::create_dir_all(parent).with_context(|| {
+                format!("failed to create ockam directory's parent at `{parent:?}`")
+            })?;
+        }
+        eprintln!("Creating ockam directory at {:?}", path);
+        std::fs::DirBuilder::new()
+            .mode(0o700)
+            .create(&path)
+            .with_context(|| {
+                format!("failed to create path to ockam data and vault at `{path:?}/vault`")
+            })?;
+    } else {
+        tracing::debug!("Ockam directory exists at {:?}.", path);
+        let verify_mode = false;
+        if verify_mode {
+            let mode = path.metadata()?.mode();
+            // Check that other users cannot modify the data inside.
+            if (mode & 0o77) != 0 {
+                eprintln!(
+                    "Ockam directory at {:?} can be read/written by other users. Restricting...",
+                    path
+                );
+                std::fs::set_permissions(&path, PermissionsExt::from_mode(0o700)).with_context(
+                    || format!("failed to update permissions for ockam data at `{path:?}`"),
+                )?;
+            }
+        }
+    }
+    Ok(path)
+}
+
+#[tracing::instrument(level = "debug", skip(data), err, fields(path = ?path))]
+pub fn write(path: &std::path::Path, data: &[u8]) -> anyhow::Result<()> {
```

Comment:
We should consider adding some authenticated encryption to files written to the disk. Having list of trusted identities and secrets in the file in plain text is the most vulnerable place here. Having symmetric key passed as environment variable can strengthen things a lot.

Reply:
I agree.